First cf-layout release

Scotchester opened the pull request:
- http://scotchester.github.io/cf-layout/docs/
- http://scotchester.github.io/cf-layout/demo/

:warning: Needs some short discussion of a few pieces before merging.

Scotchester committed:
Set up basic component build process and meta docs

Scotchester committed:
First pass at copying layout stuff from cfgov-refresh

Scotchester commented:
Notes from review:
- Once cfpb/cf-core#17 is closed, remove inline styles from `img` elements in the docs.
- Remove `content_sidebar-item` in favor of the new general-purpose `block`s with flush modifiers.

Scotchester commented:
Also, wait for cfpb/cf-core#18.

himedlooff commented on src/cf-layout.less:
```diff
@@ -0,0 +1,1248 @@
+   ========================================================================== */
+
+/* topdoc
+  name: cf-grid column dividers
```
I think it's ok to leave this here permanently under a "cf-grid helpers" section or something.

himedlooff commented:
This is looking pretty good. One observation is that I think we should consider taking `.wrapper` off of `.content_wrapper`, and make `.content_wrapper` extend or call `.wrapper();`. It looks nicer to have this:

```
.content
  .content_wrapper
    .content_main
    .content_sidebar
```

rather than this:

```
.content
  .wrapper.content_wrapper
    .content_main
    .content_sidebar
```

Scotchester committed:
Tidy up a few things

- Have `.content_wrapper` extend `.wrapper`
- Remove `.sidebar_item`; `.block`s should be used instead
- Make cf-grid helpers an official part of this component

Scotchester committed:
Update cf-core to 0.6.0

Scotchester commented:
Okiedoke, I think we're all wrapped up, here! Nice to see our first (I think) good use for `:extend`.

Scotchester committed:
Remove inline styles no longer needed

Scotchester committed:
Reposition wrapper stuff in cf-grid helpers

himedlooff commented:
Looks good thanks!
